### PR TITLE
rename the `parse()` method and adjust documentation

### PR DESCRIPTION
**Pull request by phlptp**
Based on discussions in #974

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -196,7 +196,6 @@ this:
 ```cpp
 int main(int argc, char** argv) {
     CLI::App app{"App description"};
-    argv = app.ensure_utf8(argv);
```
Comment (henryiii) [inline README.md:199]: Why is this removed? This is by far the best way to do this - `parse()` / `parse_from_cli_args()` does black magic on some operating systems, while this reliably works everywhere.

File: README.md [hunk 2]
```diff
@@ -1460,7 +1457,11 @@ itself.
 > int main() {
 >     CLI::App app;
 >     // ...
->     CLI11_PARSE(app);
```
Comment (henryiii) [inline README.md:1463]: The point of `CLI11_PARSE(one arg)` was to make this simpler to type, encouraging people to support unicode. This makes this "short" but magical method four lines long instead! It would better to just use the more reliable and safe method then (`argv = app.ensure_utf8(argv);`), which is only one line longer.

File: README.md (hunk 1 above)
Comment (phlptp) [reply README.md:199]: In this section of the documentation it is covering the most basic application,  in a vast majority of applications this is not necessary.  I can add it back if you think it valuable, but to me it creates a lot of potentially unnecessary questions when just getting started.

**Comment by phlptp** [comment]
Superseded by #987
